CorsixTH/Lua/hospital.lua

```diff
@@ -2414,3 +2414,23 @@ function Hospital:setCampaignData(campaign_data)
     self[key] = value
   end
 end
+
+--! Finds a random room with significantly long queues
+--!return room or nil The chosen room in the hospital or nil for no room busy enough
+function Hospital:getRandomBusyRoom()
+  local long_queue_rooms, active_rooms, total_queue = {}, 0, 0
+  for _, room in pairs(self.world.rooms) do
+    if room.is_active and #room.door.queue then
```

Comment:
#room.door.queue in a boolean condition is not useful, 0 is true in lua. Perhaps you didn't mean to include the # here?

Reply:
0 is acceptable in this case, it's an empty queue but a room that should be counted. I included # because it's what I need to exist, and I don't know why `if room.door.queue and #room.door.queue then` had an error.

Reply:
I'm pretty sure there's no situation where #foo returns false or nil in lua.

Reply:
Fixed.